### PR TITLE
Adaptions for iOS app

### PR DESCRIPTION
**Pull request by elliot2extreme**
- Generated icons and splash screens for iOS. Size of icon and splash master image were too small for the generation, therefor I scaled them up (dirty)
- Adaptions in src/app/app.component.ts to fix speech rate on iOS
- Adaptions in src/carrera/data-view.ts to get the app starting in iOS

### CODE REVIEW

**Comment by tkem** [comment]
I think this is a bit much for a single PR, and would be a pain to merge; as it is, it also breaks the build messing with `package.json` etc., which should be outside its scope.
Could you provide a PR with *just* the code changes? This would also be easier to review, without all the images getting in the way ;-)
Regarding images, I guess you mean something like this when running `ionic cordova resources ios`:
```
Error: encountered bad status code (400) for https://res.ionic.io/api/v1/transform
body: {"Error":"source image 2208x2208 too small for Default-Portrait@~ipadpro.png, requires at least 2048x2732 source file"}
```
As you mentioned, just scaling the images is kind of a dirty hack. I'll have a look at the masters and see what I can do next week or so.

**Comment by tkem** [comment]
You may also want to base you PR on the `develop` branch, which contains updated dependencies and will be the base for the next release (whenever *that* is).

File: src/app/app.component.ts [hunk 1]
```diff
@@ -53,6 +53,9 @@ export class AppComponent implements OnInit {
       if (readySource === 'cordova') {
         this.insomnia.keepAwake();
       }
+      if (this.platform.is('ios')) {
```
Comment (tkem) [inline src/app/app.component.ts:56]: I think this should be moved to the Speech constructor in `src/core/speech.ts`; also a comment like
```
// See https://github.com/vilic/cordova-plugin-tts/issues/40
```
would make sense for reference.

File: src/carrera/data-view.ts [hunk 2]
```diff
@@ -75,7 +75,10 @@ export class DataView {
   }
 
   static fromString(s: string) {
-    return new DataView(Uint8Array.from(s.split('').map(c => c.charCodeAt(0))).buffer);
+    // return new DataView(Uint8Array.from(s.split('').map(c => c.charCodeAt(0))).buffer);
+    // Uint8Array.from is not working in Safari at the moment
+    let array = new Uint8Array(s.split('').map(c => c.charCodeAt(0)));
```
Comment (tkem) [inline src/carrera/data-view.ts:80]: Nice catch :+1:

**Review by tkem — CHANGES_REQUESTED** [review]
PR should only include code changes, preferably as two separate commits since these are unrelated.

**Comment by elliot2extreme** [comment]
I will rework my push request to two separate commits based on the develop branch.

**Comment by tkem** [comment]
:+1:
I will take care of the image issues ASAP; watch https://github.com/tkem/openlap/issues/82

**Comment by tkem** [comment]
I had some time available, so I simply added your code changes myself to the develop branch (I'll update the changelog/readme and give contribution later, promise). Please check if this works for you now, and feel free to provide more (but maybe a little smaller) PRs if need be.

**Pull request closed without merging**